### PR TITLE
Add cluster layer for map markers.

### PR DESCRIPTION
**Pull request by tienhoah**
Rework on props management while fetching sites to map.

### CODE REVIEW

**Comment by tienhoah** [comment]
This branch created from piechart pull request, it's better to merge in that first then this.
And in order to see the effect, summary api needs to be slightly modified at the query set like so:
1. in mermaid-summary-api/src/summary_api/resources/views.py
2. in line 166, replace "return SummarySiteView.objects.filter(project_id='2c56b92b-ba1c-491f-8b62-23b1dc728890').order_by(
            'project_name', 'site_name')" 
with "return SummarySiteView.objects.all()"

I don't want to touch anything on summary api now. just test on the local as I describe above. thanks

File: src/components/LeafletMap.js [hunk 1]
```diff
@@ -40,20 +43,12 @@ const icon = L.divIcon({
...
-      geoObject.map(site => {
-        return L.marker([site.geometry.coordinates[1], site.geometry.coordinates[0]], {
-          icon: icon
-        })
-          .addTo(this.map)
-          .on('click', () => {
-            siteClickHandler(site);
-          });
-      });
+  componentDidUpdate({ markersData }) {
```
Comment (brianbancroft) [inline src/components/LeafletMap.js]: For readability, when destructuring the `prevProps` param of `componentDidUpdate`, you should incorporate the term "previous", "last", or "prior" to inform other developers that you're comparing against the last props before the component's data changed. 

For me, I'd do something like the following:
```js
componentDidUpdate({markersData: prevMarkersData}) {
  if (this.props.markersData !== prevMarkersData) {
  (...)
  }
}
```

File: src/components/LeafletMap.js [hunk 2]
```diff
@@ -62,7 +57,7 @@ class LeafletMap extends Component {
           'Map tiles by <a href="http://stamen.com">Stamen Design</a>, <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a> &mdash; Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
         subdomains: 'abcd',
         minZoom: 0,
-        maxZoom: 10,
+        maxZoom: 11,
```
Comment (brianbancroft) [inline src/components/LeafletMap.js]: Why are you allowing individuals to zoom deeper than what the tile provider allows to render?

**Comment by tienhoah** [comment]
closes #34

**Comment by brianbancroft** [comment]
Still seeing the same issue: 
image

File: src/components/LeafletMap.js [hunk 3]
```diff
@@ -71,18 +66,19 @@ class LeafletMap extends Component {
       center: [38, 16],
       zoom: 2,
       minZoom: 3,
-      maxZoom: 11,
-      zoomControl: true
+      maxZoom: 14,
```
Comment (brianbancroft) [inline src/components/LeafletMap.js]: The maximum zoom for this dataset at which you can see content from the map tiles is 13.

Comment (brianbancroft) [reply src/components/LeafletMap.js]: Correction. Looks like this is just something that applies for Null Island. In SE Asia, this goes up to 17! 

Let's do 17

**Pull request merged**